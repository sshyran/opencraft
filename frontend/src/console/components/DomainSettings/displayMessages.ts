const messages = {
    title: {
        defaultMessage: 'Domain Settings',
        description: 'Title of the page',
    },
    description: {
        defaultMessage: 'By default, your site is always accessible via an OpenCraft subdomain. Custom domains allow users to access your site via non-OpenCraft domain names. A custom domain increase your monthly OpenCraft bill by €25.',
        description: 'Page Explanation'
    },
    buttonText: {
        defaultMessage: 'Add custom domain (€25/month)',
        description: 'Button for adding custom domain'
    },
    modalTitle: {
        defaultMessage: 'Add a Custom Domain',
        description: 'Title for custom domain modal'
    },
    modalDescription: {
        defaultMessage: "Add your domain name below. If you don't have a domain yet, we recommend you go to Gandi.net to order one. Then complete this step once you have a domain.",
        description: ''
    },
    domainInput: {
        defaultMessage: 'Domain Name',
        description: 'Domain name input field'
    }
};

export default messages;
