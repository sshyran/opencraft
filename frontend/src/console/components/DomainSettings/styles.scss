@import "~styles/theme";

.domain-settings-container {
  font-family: $chivo-font;
  padding: 45px;

  .domain-settings-description {
    color: #797979;
    font-size: 16px;
    max-width: 716px;
  }

  h2 {
    color: #000000;
    font-size: 30px;
    text-align: left;
    font-weight: 600;
  }

  .addBtn {
    font-size: 14px;
  }
}

.add-domain-modal {
  padding: 10px;

  .add-domain-modal-description {
    color: #797979;
    font-size: 16px;
    max-width: 716px
  }

  h2 {
    color: #000000;
    font-size: 30px;
    text-align: left;
    font-weight: 600;
  }

  .verify-btn {
    margin-right: 0.375rem;
  }
}