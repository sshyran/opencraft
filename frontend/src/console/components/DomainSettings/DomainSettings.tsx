import * as React from 'react';
import { InstancesModel } from 'console/models';
import { ConsolePage, PreviewBox } from 'newConsole/components';
import { WrappedMessage } from 'utils/intl';
import { Button, Col, Container, Modal, Row } from 'react-bootstrap';
import { RootState } from 'global/state';
import { connect } from 'react-redux';
import messages from './displayMessages';
import './styles.scss';
import { DomainListItem } from '../DomainItem';
import { INTERNAL_DOMAIN_NAME } from 'global/constants';
import { TextInputField } from 'ui/components';
import { updateFieldValue } from 'console/actions';
interface State {
  title: string;
  subtitle: string;
  // extra state to manage the empty title and subtitle and rendering
  renderBool: boolean;
}

interface ActionProps {
  updateFieldValue: Function
}

interface StateProps extends InstancesModel { }

interface Props extends StateProps, ActionProps { }


const AddDomainButton: React.FC = () => {
  const [show, setShow] = React.useState(false);

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);
  const handleVerify = () => { console.log("verifying") }


  return (
    <div>
      <Button
        className="addBtn"
        size="lg"
        onClick={handleShow}
      >
        <WrappedMessage messages={messages} id="buttonText" />
      </Button>

      <Modal
        size={'lg'}
        show={show}
        onClose={handleClose}
        centered>
        <Modal.Body>
          <Container className="add-domain-modal">
            <h2>
              <WrappedMessage messages={messages} id="modalTitle"></WrappedMessage>
            </h2>
            <Row>
              <Col className="add-domain-modal-description">
                <p>
                  <WrappedMessage messages={messages} id="modalDescription"></WrappedMessage>
                </p>
              </Col>
            </Row>
            <Row>
              <Col lg={8}>
                <TextInputField type="domain" messages={messages} fieldName="domainInput"></TextInputField>
              </Col>
            </Row>
            <div className="flex flex-row">
              <div className="verify-btn">
                <Button variant='primary' onClick={handleVerify} disabled>Verify</Button>
              </div>
              <div>
                <Button variant='outline-primary' onClick={handleClose}>Cancel</Button>
              </div>
            </div>
          </Container>
        </Modal.Body>
      </Modal>
    </div>
  )
}

export class DomainSettingsComponent extends React.PureComponent<Props, State> {

  private deleteExternalDomain = () => {
    const instance = this.props.activeInstance;
    if (instance.data) {
      this.props.updateFieldValue(instance.data.id, "externalDomain", "")
    }
  }

  public render() {

    var subdomain;
    var externalDomain;
    var dnsConfigState;
    var externalDomainComponent;
    const instance = this.props.activeInstance;
    if (instance && instance.data) {
      subdomain = instance.data.subdomain;
      externalDomain = instance.data.externalDomain;
      dnsConfigState = instance.data.dnsConfigurationState
    }

    if (externalDomain) {
      externalDomainComponent = (
        <>
          <hr></hr>
          <DomainListItem
            domainName={externalDomain}
            isExternal={true}
            dnsState={dnsConfigState}
            onDelete={this.deleteExternalDomain} />
        </>
      )
    } else {
      externalDomainComponent = (
        <AddDomainButton></AddDomainButton>
      )
    }

    return (
      <ConsolePage
        contentLoading={false}
        showSideBarEditComponent={false}
      >
        <PreviewBox>
          <Container className="domain-settings-container">
            <h2>
              <WrappedMessage messages={messages} id="title"></WrappedMessage>
            </h2>
            <Row>
              <Col className="domain-settings-description">
                <p>
                  <WrappedMessage messages={messages} id="description"></WrappedMessage>
                </p>
              </Col>
            </Row>
            <div>
              <DomainListItem domainName={subdomain + INTERNAL_DOMAIN_NAME} isExternal={false}></DomainListItem>
              {externalDomainComponent}
            </div>
          </Container>
        </PreviewBox>
      </ConsolePage>
    );
  }
}

export const DomainSettings = connect<
  StateProps,
  ActionProps,
  {},
  Props,
  RootState
>((state: RootState) => state.console, {
  updateFieldValue
})(DomainSettingsComponent);
