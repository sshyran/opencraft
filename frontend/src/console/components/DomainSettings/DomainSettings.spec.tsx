import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { DomainSettings } from './DomainSettings';

it('renders without crashing', () => {
    const tree = setupComponentForTesting(<DomainSettings />).toJSON();
    expect(tree).toMatchSnapshot();
});
