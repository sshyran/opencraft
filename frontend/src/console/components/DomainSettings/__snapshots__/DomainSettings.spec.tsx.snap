// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="console-page"
>
  <div
    className="d-flex justify-content-center align-middle redeployment-toolbar"
  >
    <div
      className="redeployment-nav"
    >
      <span
        className="status-pill badge badge-pill badge-primary"
        style={
          Object {
            "backgroundColor": "grey",
          }
        }
      >
        <div
          className="text"
        >
          Loading
        </div>
        <i
          className="fas fa-question-circle fa-lg"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        />
      </span>
      <button
        className="float-right btn-disabled btn btn-primary btn-lg"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <div
          className="button-text-layer"
        >
          Publish changes
        </div>
      </button>
    </div>
    <div
      className="toast-container"
    />
  </div>
  <div
    className="new-console-page-container"
  >
    <div
      className="new-console-page-content row"
    >
      <div
        className="container-fluid"
      >
        <div
          className="justify-content-center row"
        >
          <div
            className="col-md-3"
          >
            <div
              className="new-customization-menu accordion"
            >
              <div
                className="card"
              >
                <div
                  className="card-header"
                >
                  <div
                    className="card-header"
                    onClick={[Function]}
                    variant="link"
                  >
                    Theme
                  </div>
                </div>
                <div
                  aria-expanded={null}
                  className="collapse show"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/theming/preview-and-colors"
                        onClick={[Function]}
                      >
                        Preview & colors
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/logos"
                        onClick={[Function]}
                      >
                        Logos
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/buttons"
                        onClick={[Function]}
                      >
                        Buttons
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/navigation"
                        onClick={[Function]}
                      >
                        Navigation
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/footer"
                        onClick={[Function]}
                      >
                        Footer
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/hero"
                        onClick={[Function]}
                      >
                        Hero
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                >
                  <div
                    className="card-header"
                    onClick={[Function]}
                    variant="link"
                  >
                    Custom Pages
                  </div>
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/custom-pages/about"
                        onClick={[Function]}
                      >
                        About
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/contact"
                        onClick={[Function]}
                      >
                        Contact
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/donate"
                        onClick={[Function]}
                      >
                        Donate
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/tos"
                        onClick={[Function]}
                      >
                        Terms of Service
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/honor"
                        onClick={[Function]}
                      >
                        Honor Code
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/privacy"
                        onClick={[Function]}
                      >
                        Privacy Policy
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                >
                  <div
                    className="card-header"
                    onClick={[Function]}
                    variant="link"
                  >
                    Instance Settings
                  </div>
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/settings/general"
                        onClick={[Function]}
                      >
                        General
                      </a>
                      <a
                        aria-current={null}
                        href="/console/settings/domain"
                        onClick={[Function]}
                      >
                        Domain
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                >
                  <div
                    className="card-header"
                    onClick={[Function]}
                    variant="link"
                  >
                    Courses (Studio)
                  </div>
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/courses/manage"
                        onClick={[Function]}
                      >
                        Manage
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col-md-7"
          >
            <div
              className="preview-box card"
            >
              <div
                className="domain-settings-container container"
              >
                <h2>
                  Domain Settings
                </h2>
                <div
                  className="row"
                >
                  <div
                    className="domain-settings-description col"
                  >
                    <p>
                      By default, your site is always accessible via an OpenCraft subdomain. Custom domains allow users to access your site via non-OpenCraft domain names. A custom domain increase your monthly OpenCraft bill by â‚¬25.
                    </p>
                  </div>
                </div>
                <div>
                  <ul
                    className="domain-list"
                  >
                    <li
                      className="list-item color-green"
                    >
                      <div
                        className="d-flex flex-wrap justify-content-between"
                      >
                        <div>
                          <div>
                            undefined.opencraft.hosting
                          </div>
                          <div
                            className="domain-type"
                          >
                            Default Subdomain
                          </div>
                        </div>
                      </div>
                    </li>
                  </ul>
                  <div>
                    <button
                      className="addBtn btn btn-primary btn-lg"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      Add custom domain (â‚¬25/month)
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
